test/index.test.ts

```diff
@@ -613,385 +613,507 @@ describe('HipThrusTS', () => {
 
         const triple = HTPipe(left, midNotCovered, rightFullyCovered);
       });
-      it('attaches properly typed data from left and right sync data attacher', async () => {
-        const aPassedIn = 'some string';
-        const bPassedIn = 4;
-        const cReturned = 6;
-        const left = {
-          attachData(context: { a: string }) {
-            expect(context.a).to.be.equal(aPassedIn);
-            return { b: bPassedIn };
-          },
-        };
 
```

Comment:
cool stuff - the cooky factory idea is working \o/
please reorganize it though - right now it's organized like:
- all the test data factories
- all the actual tests

Instead do it like this so it's easier to navigate:
test type ONE (e.g. fully covered)
-> test data factory definition
-> lifecycle one (e.g. attachData) test SYNC
-> lifecycle one (e.g. attachData) test ASYNC
-> lifecycle two (LATER - not in this PR)
-> ...
test type TWO (e.g. partially covered)
-> test data factory definition
-> ... you get the idea

- [ ] organize logically - outer structure is the PURPOSE of the test